### DISCUSSION TITLE
Unnamed RTS (Magifactory?)

### DISCUSSION THREAD

**Discussion by Dekker3D** [runevision/LayerProcGen#12] [Present your project]
This project is mostly driven by my own interests in certain systems. I'm a nerd, and proud of it; I also have ADHD and only just got on meds, at age 35. Still tweaking the dosage and specific meds for it.

So, when I noticed I was actually able to do stuff more consistently, I wanted to build something cool. I've always been fascinated with procedural animation (and I've got some decent 3D animation skills too), so that's really where this started; just a sandbox where you've got a little robot that can walk around, no matter how many legs you put on it.

My inspiration:
- Desynced, an RTS where you build whole supply lines, put bots together out of parts, and fight mean bugs. It even includes a logic scripting system. It's like Factorio, but instead of conveyor belts all items are hauled by your bots.
- Dwarf Fortress, an RTS where you don't directly control your dwarves, but instead just kinda give commands to squads and issue specific work orders. Your dwarves will happily stop working to take a break, or because there's a mean looking goblin staring at them.
- Factorio, which is a game that's even more about supply lines and less about fighting mean bugs, though it's still a bit about fighting mean bugs.

I really wanted an automation game with a magitek vibe instead of a sci-fi one, I already had some scraps of a setting for the world from another project, and I'd just seen a Twitter conversation about wheelchairs in fantasy settings (and how you might expect some kind of magical legged wheelchairs instead), so I decided to focus on golems. Basically robots, but with somewhat different mechanics, and not as perfectly controllable. The main character gets around in a golem with a seat on it, which can also be upgraded like any golem... I kinda just want to make it so you can easily move to any golem with a seat on it, and provide it with a boost of mana regen, stuff like that.

I don't really have a site for this yet, and while I have a few vids, they're extremely simple and outdated already. I could make some more when I've got my terrain looking decent and have basic combat working.

I'm playing with the thought of making it open-source, and supported by donations and selling it for a low price on Steam, with a stipulation that any forked projects must maintain a note that my project is what they forked from (which is compatible with the GPL license, as far as I understand).

I love how open-source software can just grow into something beautiful if it catches on, which is why I'm really into Godot and Blender. Cataclysm: Dark Days Ahead is also an interesting example, an open-source zombie survival Roguelike with a huge world to explore.

I... guess that's all I've got to say for now? I'm currently using Terrain3D (https://github.com/TokisanGames/Terrain3D) which isn't amazing for really large worlds (>16km), but if I replace the physics part I think I can get it to work with chunked data and shifted world origins. I'm also using AutoInject (https://github.com/chickensoft-games/AutoInject) for a form of dependency injection.

**Comment by runevision** [comment]
Thanks for sharing this about your project! It sounds very interesting and I'm looking forward to see more of it later, like screenshots or videos, once you have some that you're comfortable sharing. :)
